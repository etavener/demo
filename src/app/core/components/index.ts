export * from './card-form/card-form.component';
export * from './completed-message/completed-message.component';
export * from './payment-info/payment-info.component';
export * from './progress-display/progress-display.component';
export * from './input-expiry-date/input-expiry-date.component';
