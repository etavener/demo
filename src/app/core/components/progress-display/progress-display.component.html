<div class="progress-display">
  <div class="progress-display__label">
    Your payment is currently being Authorized
  </div>
  <div class="progress-display__spinner">
    <mat-progress-spinner mode="indeterminate"></mat-progress-spinner>
  </div>
</div>
