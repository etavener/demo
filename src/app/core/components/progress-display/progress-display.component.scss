.progress-display {
  padding: 20px;

  .progress-display__label {
    text-align: center;
    padding-bottom: 20px;
  }

  .progress-display__spinner {
    position: relative;
    width: 100px;
    height: 100px;
    margin-left: 50%;

    mat-progress-spinner {
      position: absolute;
      left: -50px;
    }
  }

}
