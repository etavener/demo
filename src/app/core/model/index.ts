export * from './payment';
export * from './worldpayResponse';
