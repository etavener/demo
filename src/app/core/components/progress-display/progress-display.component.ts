import { Component } from '@angular/core';

@Component({
  selector: 'progress-display',
  templateUrl: './progress-display.component.html',
  styleUrls: ['./progress-display.component.scss']
})
export class ProgressDisplayComponent {}
