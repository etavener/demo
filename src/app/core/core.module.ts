import { NgModule } from '@angular/core';
import {
  CardFormComponent,
  PaymentInfoComponent,
  CompletedMessageComponent,
  ProgressDisplayComponent,
  InputExpiryDateComponent,
} from './components';

import {
  DateAdapter, MAT_DATE_FORMATS, MAT_DATE_LOCALE,
  MatButtonModule,
  MatDatepickerModule,
  MatDividerModule,
  MatFormFieldModule,
  MatIconModule,
  MatInputModule,
  MatProgressSpinnerModule,
  MatSelectModule,
  MatStepperModule
} from '@angular/material';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { RouterModule} from '@angular/router';
import { STEPPER_GLOBAL_OPTIONS } from '@angular/cdk/stepper';
import { HttpClientModule } from '@angular/common/http';
import { MomentDateAdapter } from '@angular/material-moment-adapter';
import {WorldpayService} from './services/worldpay.service';
import {BrowserAnimationsModule} from '@angular/platform-browser/animations';

export const MY_FORMATS = {
  parse: {
    dateInput: 'MM/YYYY',
  },
  display: {
    dateInput: 'MM/YYYY',
    monthYearLabel: 'MMM YYYY',
    dateA11yLabel: 'LL',
    monthYearA11yLabel: 'MMMM YYYY',
  },
};

const DECLARATIONS = [
  CardFormComponent,
  PaymentInfoComponent,
  CompletedMessageComponent,
  ProgressDisplayComponent,
  InputExpiryDateComponent
];

@NgModule({
  declarations: [
    ...DECLARATIONS
  ],
  exports: [
    ...DECLARATIONS
  ],
  imports: [
    FormsModule,
    ReactiveFormsModule,
    RouterModule,
    MatStepperModule,
    MatIconModule,
    MatDatepickerModule,
    MatInputModule,
    MatFormFieldModule,
    MatProgressSpinnerModule,
    MatSelectModule,
    MatButtonModule,
    MatDividerModule,
    HttpClientModule,
    BrowserAnimationsModule
  ],
  providers: [
    WorldpayService,
    {
      provide: STEPPER_GLOBAL_OPTIONS,
      useValue: { displayDefaultIndicatorType: false }
    },
    {
      provide: DateAdapter,
      useClass: MomentDateAdapter,
      deps: [MAT_DATE_LOCALE]},
    {
      provide: MAT_DATE_FORMATS,
      useValue: MY_FORMATS
    },
  ]
})
export class CoreModule { }
