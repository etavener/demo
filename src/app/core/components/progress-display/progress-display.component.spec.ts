import { async, ComponentFixture, TestBed, } from '@angular/core/testing';
import { CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';
import { ProgressDisplayComponent } from './progress-display.component';

describe('ProgressDisplayComponent', () => {
  let component: ProgressDisplayComponent;
  let fixture: ComponentFixture<ProgressDisplayComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [ ProgressDisplayComponent ],
      schemas: [ CUSTOM_ELEMENTS_SCHEMA ]
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(ProgressDisplayComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should render the component', () => {
    expect(fixture).toMatchSnapshot();
  });
});
