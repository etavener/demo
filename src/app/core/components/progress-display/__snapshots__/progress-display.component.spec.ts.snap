// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressDisplayComponent should render the component 1`] = `
<progress-display>
  <div
    class="progress-display"
  >
    <div
      class="progress-display__label"
    >
       Your payment is currently being Authorized 
    </div>
    <div
      class="progress-display__spinner"
    >
      <mat-progress-spinner
        mode="indeterminate"
      />
    </div>
  </div>
</progress-display>
`;
